base/array.jl

```diff
@@ -1053,17 +1053,18 @@ function lexcmp(a::Array{UInt8,1}, b::Array{UInt8,1})
     return c < 0 ? -1 : c > 0 ? +1 : cmp(length(a),length(b))
 end
 
+const BitIntegerArray{N} = Union{map(T->Array{T,N}, BitInteger_types)...} where N
 # use memcmp for == on bit integer types
-function ==(a::Array{T,N}, b::Array{T,N}) where T<:BitInteger where N
+function ==(a::A, b::A) where A <: BitIntegerArray
```

Comment:
So `T` was looking like an`eltype`, so I chose `Arr`; if still not good, please suggest.